src/test/java/com/qunite/api/web/AuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,231 @@
+class AuthenticationIntegrationTest {
+
+  private final String url = "auth";
+
+  @Autowired
+  private MockMvc mockMvc;
+
+  @Autowired
+  private UserMapper userMapper;
+
+  @Autowired
+  private UserService userService;
+
+  @Autowired
+  private UserRepository userRepository;
+
+  @AfterEach
+  public void cleanAll() {
+    userRepository.deleteAll();
+  }
+
+  @Test
+  void unauthenticatedUserShouldAccessEndpoints() throws Exception {
+    mockMvc.perform(post("/{url}/sign-up", url))
+        .andExpect(status().isBadRequest());
+    mockMvc.perform(post("/{url}/sign-in", url))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  void signUpShouldCreateNewUserWithValidData() throws Exception {
+    var user = new User();
+    user.setUsername("John");
+    user.setPassword("Johnson");
+    user.setEmail("john@gmail.com");
+
+    var dto = userMapper.toUserCreationDto(user);
+    var json = new ObjectMapper().writeValueAsString(dto);
+
+    mockMvc.perform(
+            post("/{url}/sign-up", url)
+                .contentType(MediaType.APPLICATION_JSON).content(json))
+        .andExpect(status().is2xxSuccessful());
+
+    assertThat(userService.findByUsername("John")).isPresent();
+  }
+
+  @Test
+  @Sql("/users-create.sql")
+  void signUpShouldNotCreateWithExistingUsername() throws Exception {
+    var user = new User();
+    user.setUsername("First");
+    user.setPassword("Johnson");
+    user.setEmail("john@gmail.com");
+
+    var dto = userMapper.toUserCreationDto(user);
+    var json = new ObjectMapper().writeValueAsString(dto);
+
+    mockMvc.perform(
+            post("/{url}/sign-up", url)
+                .contentType(MediaType.APPLICATION_JSON).content(json))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  @Sql("/users-create.sql")
+  void signUpShouldNotCreateWithExistingEmail() throws Exception {
+    var user = new User();
+    user.setUsername("John");
+    user.setPassword("Johnson");
+    user.setEmail("User1@user.com");
+
+    var dto = userMapper.toUserCreationDto(user);
+    var json = new ObjectMapper().writeValueAsString(dto);
+
+    mockMvc.perform(
+            post("/{url}/sign-up", url)
+                .contentType(MediaType.APPLICATION_JSON).content(json))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  void signUpShouldEncryptPassword() throws Exception {
+    var user = new User();
+    user.setUsername("John");
+    user.setPassword("Johnson");
+    user.setEmail("john@gmail.com");
+
+    var dto = userMapper.toUserCreationDto(user);
+    var json = new ObjectMapper().writeValueAsString(dto);
+
+    mockMvc.perform(
+            post("/{url}/sign-up", url)
+                .contentType(MediaType.APPLICATION_JSON).content(json))
+        .andExpect(status().isOk());
+
+    var createdUser = userService.findByUsername("John").get();
+    var encoder = new BCryptPasswordEncoder();
+    assertTrue(encoder.matches("Johnson", createdUser.getPassword()));
+  }
+
```

Comment:
why you create separate test for encoding? u can check password encoding in signUpShouldCreateNewUserWithValidData() method

Reply:
If the signUpShouldCreateNewUserWithValidData is incorrect, there can be many reasons for that, this method specifies and check certain reason, so we can find out if this is a proplem.